client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+    const CommentDecorator = useMemo(() => getCommentDecorator(commentApp), [commentApp])
+    const comments = useSelector(commentsSelector, shallowEqual);
+    const enabled = useSelector(commentApp.selectors.selectEnabled);
+    const focusedId = useSelector(commentApp.selectors.selectFocused);
+
+    const ids = useMemo(() => comments.map(comment => comment.localId), [comments]);
+
+    const commentStyles = useMemo(() => ids.map(id => {return {
+      type: COMMENT_STYLE_IDENTIFIER + id,
+      style: enabled ? {
+        'background-color': (focusedId !== id) ? '#01afb0' : '#007d7e'
+      } : {}
+    }}), [ids, enabled]);
+
+    const [uniqueStyleId, setUniqueStyleId] = useState(0)
+
+    const previousFocused = usePrevious(focusedId);
+    const previousIds = usePrevious(ids);
+    const previousEnabled = usePrevious(enabled);
+    useEffect(() => {
+      // Only trigger a focus-related rerender if the current focused comment is inside the field, or the previous one was
+      const validFocusChange = previousFocused !== focusedId && (previousIds && previousIds.includes(previousFocused) || ids.includes(focusedId))
+
+      if ((!validFocusChange) && previousIds === ids && previousEnabled === enabled) {
+        return
+      }
+
+      // Filter out any invalid styles - deleted comments, or now unneeded STYLE_RERENDER forcing styles
+      const filteredContent = filterInlineStyles(inlineStyles.map(style => style.type).concat(ids.map(id => COMMENT_STYLE_IDENTIFIER + id)), editorState.getCurrentContent())
+      // Force reset the editor state to ensure redecoration, and apply a new (blank) inline style to force inline style rerender
+      // This must be entirely new for the rerender to trigger, hence the unique style id, as with the undo stack we cannot guarantee
+      // that a previous style won't persist without filtering everywhere, which seems a bit too heavyweight
+      // This hack can be removed when draft-js triggers inline style rerender on props change
+      setEditorState(editorState => forceResetEditorState(editorState, 
+        Modifier.applyInlineStyle(
+          filteredContent,
+          getFullSelectionState(filteredContent),
+          'STYLE_RERENDER_'+uniqueStyleId
+        )
+      ))
+      setUniqueStyleId((id) => (id + 1) % 200);
+    }, [focusedId, enabled, inlineStyles, ids, editorState])
+
+    const timeoutRef = useRef();
+    useEffect(() => {
+      window.clearTimeout(timeoutRef.current);
+      const filteredEditorState = EditorState.push(
+        editorState,
+        filterInlineStyles(inlineStyles.map(style => style.type), editorState.getCurrentContent())
+      );
+      timeoutRef.current = window.setTimeout(
+        onSave(serialiseEditorStateToRaw(filteredEditorState)),
+        250,
+      );
+      return () => {
+        window.clearTimeout(timeoutRef.current);
+      }
+    }, [editorState, inlineStyles]);
```

Comment:
I don't think we'd need it directly in Draftail, since Draftail is doing a sensible thing here: it's filtering to the inline styles it knows about, it's just we don't want it to save all of those legitimate styles because some are comments. This is probably also where we'd want to put the comment position updating logic, when we serialise it. So yep, will add a comment to that effect!